drivers/tty/serdev/pibridge.c

```diff
@@ -400,34 +402,45 @@ int pibridge_req_io(u8 addr, u8 cmd, u8 *snd_buf, u16 snd_len, u8 *rcv_buf,
 			"receive head error in io-req\n");
 		return -EIO;
 	}
+	crc = pibridge_crc8(0, (u8 *) &pkthdr, sizeof(pkthdr));
 
-	if (rcv_len != 0) {
-		if (pibridge_recv(rcv_buf, rcv_len) != rcv_len) {
+	to_receive = min((u16) pkthdr.len, rcv_len);
+	to_discard = pkthdr.len - to_receive;
+
+	if (to_receive) {
+		if (pibridge_recv(rcv_buf, to_receive) != to_receive) {
 			dev_warn_ratelimited(&pibridge_s->serdev->dev,
-				"receive data error in io-req\n");
-			return -EBADMSG;
+				"receive data error in io-req(len:%d)\n",
+				to_receive);
+			return -EIO;
 		}
 	}
 
+	if (to_discard) {
+		/* The provided buffer was too small. Discard the rest of the
+		   received data as well as the following CRC checksum byte. */
```

Comment:
Nit: Code comment doesn't adhere to <a href="https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/process/coding-style.rst#n619">kernel coding style</a>.